tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -857,6 +857,43 @@ var _ = Describe("odo devfile push command tests", func() {
 
 		})
 	})
+	Context("Verify source code sync location", func() {
+
+		It("Should sync to the correct dir in container if project is present", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+
+			// devfile with clonePath set in project field
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-projects.yaml"), filepath.Join(context, "devfile.yaml"))
+			// we do not need to check source code location inside the container, as the expected workingDir
+			// is already set in devfile.yaml, odo push would fail if it is not synced to correct location
```

Comment:
I think `odo push` doesn't fail in case of nodejs components even if there is no package.json. I feel it's better to check for the presence of some files at the source location.

Reply:
Done.